drivers/flash/flash_stm32_qspi.c

```diff
@@ -639,9 +840,66 @@ static int spi_nor_process_bfp(const struct device *dev,
 
 	LOG_DBG("Page size %u bytes", data->page_size);
 	LOG_DBG("Flash size %u bytes", flash_size);
+
+	addr_mode = jesd216_bfp_addrbytes(bfp);
+	if (addr_mode == JESD216_SFDP_BFP_DW1_ADDRBYTES_VAL_3B4B) {
+		struct jesd216_bfp_dw16 dw16;
+
+		if (jesd216_bfp_decode_dw16(php, bfp, &dw16) == 0) {
+			/*
+			 * According to JESD216, the bit0 of dw16.enter_4ba
+			 * portion of flash description register 16 indicates
+			 * if it is enough to use 0xB7 instruction without
+			 * write enable to switch to 4 bytes addressing mode.
+			 */
+			if (dw16.enter_4ba & 0x1) {
+				rc = qspi_program_addr_4b(dev);
+				if (rc == 0) {
+					data->flag_access_32bit = true;
+					LOG_INF("Flash - address mode: 4B");
+				} else {
+					LOG_ERR("Unable to enter 4B mode: %d\n",
+						rc);
+					return rc;
+				}
+			}
+		}
+	}
+	/*
+	 * Only check if the 1-4-4 (i.e. 4READ) fast read operation is
+	 * supported - other modes - e.g. 1-1-4 (QREAD) or 1-1-2 (DREAD) are
+	 * not.
+	 */
+	struct jesd216_instr res;
+
+	rc = jesd216_bfp_read_support(php, bfp, JESD216_MODE_144, &res);
+	if (IS_ENABLED(STM32_QSPI_QUAD_IO) && rc > 0) {
+		/* Program flash memory to use SIO[0123] */
+		qspi_program_quad_io(dev);
+
+		LOG_INF("Mode: 1-4-4 with instr:[0x%x] supported!", res.instr);
```

Comment:
You can have success info even if the `qspi_program_quad_id` failed, and the `aspi_program_quad_id` has similar log at the end `https://github.com/zephyrproject-rtos/zephyr/pull/40859/files#diff-39156d185a094793a5d212e3ee4bbc9d275cf3baef6c5eca6aa95089521d1821R804` making this line redundant.

Reply:
There is a difference between those two `LOG_INF()` information.

The one with "Mode: 1-4-4..." informs about the instruction code supported by the flash memory.
The one from the pasted above link informs that in SPI-NOR memory the QSPI mode has been enabled (the status register has correct bit set).